pkgs/by-name/xe/xen_4_18/package.nix

```diff
@@ -0,0 +1,48 @@
+{
+  buildXenPackage,
+  ocaml-ng,
+  slim ? false,
+}:
+
+buildXenPackage.override
+  {
+    inherit slim;
```

Comment:
Perhaps this should be part of the main parameters rather than something you have to `.override`?

Reply:
I'm not exactly sure what's the best way to do this, tbh. I need users to be able to override the options in the generic builder, but they can't to `xen.override` because that can only override one input: `xen_4_19`, and they can't do `xen_4_19.override` because that can only override `buildXenPackage`, `ocaml-ng` and `slim`. How do I make sure overrides go all the way to the generic builder, and don't try to override each individual expression?

Reply:
Yeah, I think you’d need to duplicate all that stuff.

My suggestion is to make `pkgs/by-name/xe/xen/package.nix` that contains all the feature flags explicitly and `xenVersion ? "4.19"`, which it uses to import e.g. `./4.18.nix` containing the version‐specific stuff. Then you can do `xen_4_18 = callPackage ../by-name/xe/xen/package.nix { xenVersion = "4.18"; };` in `all-packages.nix`, which is explicitly allowed, and it should work with all the override flags.

This isn’t the most elegant thing in the world, but I think it’s the best way to achieve multi‐version packages like this within the current `by-name` structure that follows the letter of the law and isn’t too hacky.

Reply:
I really would rather avoid adding anything to `all-packages.nix`. Maybe we should copy all of the feature flags to `4.1{8,9}/package.nix` and that would allow users to override `xen_4_1X`? Not sure about the unpinned `xen` though.

Reply:
I believe it's ok to use override for isSlim, that's how users expect to override xen, without reading all of those builder implementation details

Reply:
Yeah, people who need more granularity can write their own Xen derivations.

Reply:
Unfortunately I don’t think we have a good way to do multi‐version packages that doesn’t involve duplication or putting things into `all-packages.nix` at present. `by-name` is designed for the common case and will probably need extension to handle this sort of thing elegantly. But if you really don’t want anything in `all-packages.nix`, even though it’s explicitly okay to use `by-name` stuff like that, then duplication should work.

Note that moving the QEMU stuff into `by-name` poses the same problem for overrides there.

Reply:
I don’t have enough self‐control to stop myself pointing out that dropping the non‐slim Xen would happen to conveniently solve this quandary :)

Reply:
I really don't think we should remove fat Xen at this time.